packages/api/src/taskInfo.ts

```diff
@@ -16,27 +16,23 @@
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
 
-export type TaskState = 'running' | 'completed';
-type TaskStatus = 'in-progress' | 'success' | 'failure';
+export type TaskState = 'loading' | 'success' | 'error';
 
-export interface Task {
+export interface NotificationTaskInfo extends TaskInfo {
+  body: string;
+  markdownActions?: string;
+  // overwrite some fields type
+  state: 'success';
+  progress: undefined;
+  error: undefined;
```

Comment:
here I don't see why we set undefined, AFAIK it's optional in the parent interface so it's already undefined

Reply:
We want to limit the value possible, as NotificationTaskInfo extend TaskInfo, not having this could let someone set an error or a progress value, by enforcing undefined we prevent a value to be set

Reply:
ok

Reply:
in this case should it be using `Omit` ? to really don't see these 2 fields

Reply:
Oh yeah, good idea